CMakeLists.txt

```diff
@@ -370,9 +368,8 @@ if(CMAKE_CUDA_COMPILER)
   message(STATUS "Cuda language found. Will build NVIDIA Platform.")
 endif()
 
-# Check for CURL and OpenSSL
-find_package(CURL COMPONENTS HTTP HTTPS QUIET) 
-find_package(OpenSSL QUIET)
+# Check for OpenSSL
+find_package(OpenSSL REQUIRED)
```

Comment:
This implies that we now want the RestClient to always be built. If we have a simulation-only user, they might want to be able to build the stack without needing remote qpu invocation.

Reply:
Alright, I'll remove it. I understand the convenience. At the same time, I think it is a bit too easy to accidentally remove part of the compilation without it being caught by CI. I think we need to think about a way to have the CI fail if parts that shouldn't be disabled are disabled.